ERCS/erc-7656.md

```diff
@@ -0,0 +1,279 @@
+          revert(0x1c, 0x04)
+        }
+
+      // Store account address in memory before salt and chainId
+        mstore(0x6c, deployed)
+
+      // Emit the Created event
+        log4(
+          0x6c,
+          0x60,
+          0xc6989e4f290074742210cbd6491de7ded9cfe2cd247932a53d31005007a6341a,
+          implementation,
+          tokenContract,
+          tokenId
+        )
+
+      // Return the account address
+        return(0x6c, 0x20)
+      }
+
+    // Otherwise, return the computed account address
+      mstore(0x00, shr(96, shl(96, computed)))
+      return(0x00, 0x20)
+    }
+  }
+
+  function compute(
+    address implementation,
+    bytes32 salt,
+    uint256 /* chainId */,
+    address /* tokenContract */,
+    uint256 /* tokenId */
+  ) external view override returns (address) {
+    // solhint-disable-next-line no-inline-assembly
+    assembly {
+    // Copy bytecode + constant data to memory
+      calldatacopy(0x8c, 0x24, 0x80) // salt, chainId, tokenContract, tokenId
+      mstore(0x6c, 0x5af43d82803e903d91602b57fd5bf3) // ERC-1167 footer
+      mstore(0x5d, implementation) // implementation
+      mstore(0x49, 0x3d60ad80600a3d3981f3363d3d373d3d3d363d73) // ERC-1167 constructor + header
+
+    // Copy create2 computation data to memory
+      mstore8(0x00, 0xff) // 0xFF
+      mstore(0x35, keccak256(0x55, 0xb7)) // keccak256(bytecode)
+      mstore(0x01, shl(96, address())) // registry address
+      mstore(0x15, salt) // salt
+
+    // Store computed account address in memory
+      mstore(0x00, shr(96, shl(96, keccak256(0x00, 0x55))))
+
+    // Return computed account address
+      return(0x00, 0x20)
+    }
+  }
+
+  /// @dev Returns true if interfaceId is IERC7656Registry's interfaceId
+  /// This contract does not explicitly extend IERC165 to keep the bytecode as small as possible
+  function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
+    return interfaceId == 0xc6bdc908;
+  }
+}
+```
+
+A simple implementation of `IERC7656Contract`:
+
+```solidity
+contract ERC7656Contract is IERC7656Contract {
+
+  function token() public view virtual returns (uint256, address, uint256) {
+    bytes memory footer = new bytes(0x60);
+     assembly {
+      extcodecopy(address(), add(footer, 0x20), 0x4d, 0x60)
+    }
+    return abi.decode(footer, (uint256, address, uint256));
+  }
+}
+```
+
+## Rationale
+
+The expansion of the registry's capabilities to manage contracts beyond accounts provides several advantages:
+
+- **Flexibility**: Developers can allow NFTs to interact with a broader range of linked contracts, unlocking new use cases and functionalities.
+- **Compatibility**: By ensuring that account-like contracts can still be identified as such, the proposal maintains backward compatibility with [ERC-6551](./eip-6551.md).
+- **Innovation**: This proposal encourages further innovation in the NFT space by removing limitations on the types of contracts that can be associated with NFTs.
```

Comment:
This section should describe technical choices made within the proposal itself. This reads more like Motivation because it's comparing this proposal to another.

Reply:
I moved this to the Motivations.